### ISSUE TITLE
Improve link in `MetricNumber`'s tooltip.

### ISSUE THREAD

**Issue by tomalec** [woocommerce/google-listings-and-ads#806]
_This is a follow-up from https://github.com/woocommerce/google-listings-and-ads/pull/749_

#### Nested link
When [we added links](https://github.com/woocommerce/google-listings-and-ads/pull/749/files/979e77b590c59a8f81e0f799e34c2fe8fc4e9486..d5d56b60becb9bd6d59ec7da0d92c0ebbd47304d#diff-b0d6e30ff26f71b9f5f6d1f961c1e619b333f8415b5e18dc92b8e6d8813a2838R91-R108) inside `MetricNumber`'s `Tooltip`, we created invalid HTML - `<a>…<a>…</a>…</a>`. Per HTML spec, anchors should not contain any interactive content, so should not include nested links.

I'm not sure if it produces any observable, bug other than console warning, but I guess that may be a reason of some effect like the one described at https://github.com/woocommerce/google-listings-and-ads/pull/749#issuecomment-859218396:

> Just one small thing to note: in the tooltip popup, if we click on the black popup, it would act like we are clicking the element at the back and causes the tab click / tab change. 

I believe we could solve that by changing `SummaryNumber` to be a button, not a link (use `onClick` instead of `href`), then making the tooltip contain a simple link.

##### Additional details:

I'm not sure if it's worth resolving now and that way, as the problem applies only to a subset of `MetricNumber`s, and probably will eventually vanish once the Google API will get more stable.

#### ARIA label

Currently, we duplicate the text for ARIA label, even though the content is rendered in the DOM (it's not an image that we need to describe). In my opinion, in an ideal world `Tooltip` should have an option to point to its content by `labeledby`, so we as the consumers should not have to bother about it.

Actually, I think `describedby` could fit here better. Per [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute):
> `aria-labelledby` is very similar to `aria-describedby`: A label provides essential information about an object, while a description provides extended information that the user might need.

So we could have:
- (i).label - "This data is currently available for paid campaigns only."
- (i).description - "This data is currently available for paid campaigns only. Please try again later, or go to <linkOfGoogleMerchantCenter /> to track your performance for Google Free Listings."

##### Additional details:
I guess changing from `label` to `labeledby` requires some work in `@woocommerce/components` and I'm not sure if we plan to focus on improving DevX for A11Y.

Speaking of adding `description` to the existing label (assuming `Tooltip` does not do that for us)
it seems like a pretty cosmetic change as the information is already there.

//cc @eason9487

**Comment by eason9487** [comment]
> When [we added links](https://github.com/woocommerce/google-listings-and-ads/pull/749/files/979e77b590c59a8f81e0f799e34c2fe8fc4e9486..d5d56b60becb9bd6d59ec7da0d92c0ebbd47304d#diff-b0d6e30ff26f71b9f5f6d1f961c1e619b333f8415b5e18dc92b8e6d8813a2838R91-R108) inside `MetricNumber`'s `Tooltip`, we created invalid HTML - `<a>…<a>…</a>…</a>`. Per HTML spec, anchors should not contain any interactive content, so should not include nested links.

I have looked into the current version of `SummaryNumber` and `Tooltip`'s implementations, but there is no ideal way to fix this issue. Using `onToggle` instead of `href` will change the layout and style of `SummaryNumber`. It might be the closest way but needs to override many CSS.

Other workarounds such as using **React Portal**, hijacking the click event from the topper layer, or replacing it with `Popover` are also not ideal due to many re-implementation or being too hacky. Probably the better way would be to add more flexible support to the upstream component(s).

**Comment by eason9487** [comment]
Since [WordPress 6.4](https://github.com/WordPress/gutenberg/blob/wp/6.4/packages/components/CHANGELOG.md#2580-2023-09-20), the `Tooltip` component provided by `@wordpress/components` was refactored to use `ariakit`, whose `Ariakit.Tooltip` component content will be mounted on the `<body>` element. Therefore, this issue will be automatically resolved as the minimum supported version reaches WordPress 6.4.

Ref: https://github.com/WordPress/gutenberg/pull/48440

It can be checked by running `document.querySelector('a a')` on DevTool Console when the tooltip is showing.

![Image](https://github.com/user-attachments/assets/4ecbea22-493d-4b98-9c84-4d68d3634f98)
